test/test_cuda.py

```diff
@@ -5086,6 +5086,43 @@ def test_memory_profiler_viz(self):
         self.assertTrue("test_memory_profiler_viz" in plot)
         self.assertTrue('category' in plot)
 
+    def test_cycles(self):
+        fired = False
+
+        def observer(html):
+            nonlocal fired
+            fired = True
+            self.assertTrue('torch.Tensor' in html)
+            self.assertTrue('test_cuda' in html)
+            self.assertTrue('cell_contents' in html)
+
+        disarm = observe_tensor_cycles(observer)
+
+        def noop():
+            pass
+
+        try:
+            def create():
+                x = torch.empty(3, 4, device='cuda')
+
+                def foo(p):
+                    if p:
+                        return foo(not p)
+                    else:
+                        return x
+                return foo
+            create()
+            gc.collect()
+            # the callback has to run outside of the collect
+            # call so it doesn't actual fire until the next
+            # method call after a gc.collect
+            noop()
+            self.assertTrue(fired)
+        finally:
+            disarm()
```

Comment:
Does having disarm allow us to toggle this on and off dynamically? By calling these two apis?
```
disarm = observe_tensor_cycles(observer)
disarm()
```